File: services/order-service/src/main/java/com/bookland/order/entity/ShippingMethod.java
package com.bookland.order.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "shipping_method")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ShippingMethod {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(nullable = false)
    @Builder.Default
    private Double price = 0.0;
}

File: BookLand_FE/src/types/ShippingMethod.ts
import type { Bill } from './Bill';

export interface ShippingMethod {
    id: number;
    name: string;
    description?: string;
    price: number;
    bills?: Bill[];
}

export interface ShippingMethodRequest {
    name: string;
    description?: string;
    price: number;
}
